gems/smithy-client/lib/smithy-client/auth_scheme.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # Base class for all AuthScheme classes.
+    class AuthScheme
```

Comment:
Rather than a class - could we have a map within a module? If we need to reference auth schemes for resolution. I'm trying to think of ways to simplify things.

Reply:
I think a class is preferred right now, or for it to be API private if possible..